functionary/ui/views/scheduling.py

```diff
@@ -0,0 +1,285 @@
+        return HttpResponseForbidden()
+
+    if (function_id := request.GET.get("function", None)) in ["", None]:
+        return HttpResponse("No function selected.")
+
+    func = get_function(function_id, env)
+    if isinstance(func, HttpResponseNotFound):
+        return func
+
+    """
+    If the request includes the scheduled_task_id, substitute the default
+    widget values for the existing ScheduledTask parameters.
+    """
+    existing_parameters = None
+    if (scheduled_task_id := request.GET.get("scheduled_task_id", None)) is not None:
+        scheduled_task = ScheduledTask.objects.get(id=scheduled_task_id)
+        existing_parameters = scheduled_task.parameters
+
+    parameters = get_parameters(func, parameter_values=existing_parameters)
+    context = {"parameters": parameters}
+    return render(request, "partials/function_parameters.html", context)
+
+
+@require_POST
+@login_required
+def crontab_minute_param(request: HttpRequest) -> HttpResponse:
+    if not is_valid_scheduled_minute(request.POST.get("scheduled_minute")):
+        return render(request, "partials/crontab_invalid.html", {"field": "Minute"})
+    return HttpResponse("")
+
+
+@require_POST
+@login_required
+def crontab_hour_param(request: HttpRequest) -> HttpResponse:
+    if not is_valid_scheduled_hour(request.POST.get("scheduled_hour")):
+        return render(request, "partials/crontab_invalid.html", {"field": "Hour"})
+    return HttpResponse("")
+
+
+@require_POST
+@login_required
+def crontab_day_of_week_param(request: HttpRequest) -> HttpResponse:
+    if not is_valid_scheduled_day_of_week(request.POST.get("scheduled_day_of_week")):
+        return render(
+            request, "partials/crontab_invalid.html", {"field": "Day of week"}
+        )
+    return HttpResponse("")
+
+
+@require_POST
+@login_required
+def crontab_day_of_month_param(request: HttpRequest) -> HttpResponse:
+    if not is_valid_scheduled_day_of_month(request.POST.get("scheduled_day_of_month")):
+        return render(
+            request, "partials/crontab_invalid.html", {"field": "Day of month"}
+        )
+    return HttpResponse("")
+
+
+@require_POST
+@login_required
+def crontab_month_of_year_param(request: HttpRequest) -> HttpResponse:
+    if not is_valid_scheduled_month_of_year(
+        request.POST.get("scheduled_month_of_year")
+    ):
+        return render(
+            request, "partials/crontab_invalid.html", {"field": "Month of year"}
+        )
+    return HttpResponse("")
+
+
+def _create_scheduled_task(request: HttpRequest, data: dict, task_params: dict):
+    """Helper function for creating scheduled task"""
+    scheduled_task = ScheduledTask.objects.create(
+        name=data["name"],
+        environment=data["environment"],
+        description=data["description"],
+        function=data["function"],
+        parameters=task_params,
+        creator=request.user,
+    )
+
+    scheduled_task.periodic_task = create_periodic_task(data, scheduled_task)
```

Comment:
The `create_periodic_task` call also sets the value on the `scheduled_task` and saves it, so it doesn't seem like you have to actually assign it here, not that it hurts anything.

I might have structured the helpers a little different such that you just have a single call that creates the scheduled task and the periodic task for you, associating them properly etc. in a single transaction.  But that's not something I feel strongly enough about to say it needs to change here.

Reply:
Fixed. I think in the future, I could probably just move the `create_periodic_task` to the `ScheduledTask` model as a helper function, so I don't need to pass the `ScheduledTask` object as reference